src/Components/WebView/WebView/src/IpcReceiver.cs

```diff
@@ -89,7 +91,14 @@ private Task DispatchBrowserEventAsync(PageContext pageContext, string eventDesc
         {
             var renderer = pageContext.Renderer;
             var jsonSerializerOptions = pageContext.JSRuntime.ReadJsonSerializerOptions();
-            var webEventData = WebEventData.Parse(renderer, jsonSerializerOptions, eventDescriptor, eventArgs);
+            // JsonSerializerOptions are tightly bound to the JsonContext. Cache it on first use using a copy
+            // of the serializer settings.
+            if (_jsonContext is null)
+            {
+                _jsonContext = new(new JsonSerializerOptions(jsonSerializerOptions));
```

Comment:
Same here about moving the `ReadJsonSerializerOptions` call.

Reply:
```suggestion
                var jsonSerializerOptions = pageContext.JSRuntime.ReadJsonSerializerOptions();
                _jsonContext = new(new JsonSerializerOptions(jsonSerializerOptions));
```